src/Application.php

```diff
@@ -16,7 +16,7 @@ class Application extends Console\Application
      */
     protected function getVersionInDir($directory)
     {
-        if (!$directory) {
+        if (!$directory || dirname($directory) === $directory) {
```

Comment:
Would this `dirname($directory) === $directory` check be better at the end of this method? I can’t imagine why you’d ever run cow from the root of your filesystem, but this could break it 😉

Reply:
This doesn't "break" anything, it just means the version number isn't resolved, it's mostly for aesthetics of the output than a critical cow feature.

I had put it at the end originally but decided to just slide it up to the top with the existing condition to stop duplication.

Reply:
Oh, and other parts of cow would break sooner if the bin file was literally located in the system root (because it depends on looking in the parent dir to find the autoload file (https://github.com/silverstripe/cow/blob/2.2.0/bin/cow#L9)